Marlin/src/lcd/extui/ui_api.cpp

```diff
@@ -260,7 +260,7 @@ namespace ExtUI {
   #ifdef TOUCH_UI_LCD_TEMP_SCALING
     #define GET_TEMP_ADJUSTMENT(A) (float(A) / (TOUCH_UI_LCD_TEMP_SCALING))
   #else
-    #define GET_TEMP_ADJUSTMENT(A) A
+    #define GET_TEMP_ADJUSTMENT(A) float(A)
```

Comment:
When most of the world (including me) is rushing towards informal scripting languages and simplified code style, why is it preferable to make this class require an explicit cast, rather than permitting the class to "do the right thing"?

Philosophically speaking, should we continue to build the code with more such 'caveats' — calling for an explicit cast for certain cases that "you'll need to know about before you enter" …?… or should we aim to make the code more non-pedantic, casual, and "script-like" to conceal the ugliness of its C++ underpinnings? I may be in a minority, but I have come to see C++ as a language designed to hide the C++ so that you can come up with your own 'super-language' on top of C++ while still producing optimal binary code in the end. Whether merited or not, changes like this feel like a move in the opposite direction.

Sure, we also see Javascript programmers running to TypeScript because they crave more formal and strict typing, and that makes sense too. What I hope for is something in the middle, where we prevent coders from doing "dumb" things but if someone forgets to `float()` the thing being assigned to a `float` the compiler will still be permissive and "do the right thing" with it.

Reply:
Ok. I've heard your complaint here. I'm almost finished with my changes so I'll switch back to implicit float casting so these are not required anymore.